File: Objective-C Sample Code/Chapter 7/Top10/Top10 WatchKit App Extension/DetailController.m
//
//  DetailController.m
//  Top10
//
//  Created by Cory Bohon on 12/11/16.
//  Copyright © 2016 Cory Bohon. All rights reserved.
//

#import "DetailController.h"

@implementation DetailController

- (void)awakeWithContext:(id)context
{
    [super awakeWithContext:context];
    
    JSONItem *detailItem = (JSONItem *)context;
    [self.songTitleLabel setText:detailItem.songTitle];
    
    [self.audioPreviewMovie setMovieURL:detailItem.audioPreviewURL];
    [self.audioPreviewMovie setPosterImage:[WKImage imageWithImageData:[NSData dataWithContentsOfURL:detailItem.imageURL]]];
    [self.audioPreviewMovie setLoops:NO];
}

@end

File: Swift Sample Code/Chapter 6/Top10/Top 10 WatchKit App Extension/DetailController.swift
//
//  DetailController.swift
//  Top10
//
//  Created by Cory Bohon on 2/14/17.
//  Copyright © 2017 Cory Bohon. All rights reserved.
//

import UIKit
import WatchKit

class DetailController: WKInterfaceController {
    
    @IBOutlet weak var songTitleLabel: WKInterfaceLabel?
    @IBOutlet weak var audioPreviewMovie: WKInterfaceMovie?

    override func awake(withContext context: Any?) {
        super.awake(withContext: context)
        
        let detailItem: JSONItem = context as! JSONItem
        
        self.songTitleLabel?.setText(detailItem.songTitle)
        self.audioPreviewMovie?.setMovieURL(detailItem.audioPreviewURL)
        
        do {
            try self.audioPreviewMovie?.setPosterImage(WKImage(imageData: Data(contentsOf: detailItem.imageURL)))
        } catch _ {
            print("Unable to fetch image")
        }
        
        self.audioPreviewMovie?.setLoops(false)
    }
    
}
